CONTRIBUTING.md

```diff
@@ -39,8 +39,40 @@ Or to be safe, you call apply all the required resources at once:
 
 1. `k-apply-all`
 
+### Releasing a New Version
+
+1. Create a new release branch prefixed with `release-` followed by the version number and push the branch to GitHub.
+
+   ```shell
+    git checkout -b release-0.0.1
```

Comment:
release branches should only be created for major/minor releases, not for patch releases. You can see this distinction in the two different processes for core crossplane:

* major/minor release: https://github.com/crossplane/release/blob/main/.github/ISSUE_TEMPLATE/release.md
  * create a new release branch
* patch releases: https://github.com/crossplane/release/blob/main/.github/ISSUE_TEMPLATE/patch_release.md
  * reuse existing release branch, backport fixes to the existing major/minor branch.

Reply:
we typically also just create this branch in the github UI - could be an easier way to do it 😇